File: poo/python/Interfaces/Carbono/CarbonFootprintApp.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 12/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

from Bicycle import Bicycle
from Building import Building
from Car import Car

if __name__ == '__main__':

    objetos = []

    # Objetos Car #
    objetos.append(Car())
    objetos.append(Car('Uno', 'Gasolina', 1200, 5))

    # Objetos Building #
    objetos.append(Building())
    objetos.append(Building(12, True, 36, True))

    # Objetos Bicycle #
    objetos.append(Bicycle())
    objetos.append(Bicycle('Velha', 'Ferrugem, Pneu furado'))

    for i in range(6):
        if i == 2 or i == 4:
            print()
        print(objetos[i])

File: poo/java/Interfaces/Carbono/CarbonFootprintApp.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 04/04/14
 *  R.A.: 86492
 */

public class CarbonFootprintApp{
    public static void main(String args[]){
        CarbonFootprint objetos[] = new CarbonFootprint[6];
        int i;

        /* Objetos Car */
        objetos[0] = new Car();
        objetos[1] = new Car("Uno", "Gasolina", 1200, 5);

        /* Objetos Building */
        objetos[2] = new Building();
        objetos[3] = new Building(12, true, 36, true);

        /* Objetos Bicycle */
        objetos[4] = new Bicycle();
        objetos[5] = new Bicycle("Velha", "Ferrugem, Pneu furado");

        /* Lista as pegadas de carbono */
        for(i = 0; i < 6; i++){
            if(i == 2 || i == 4) System.out.println("\n");
            System.out.println(objetos[i]);
        }

    }
}
